tests/elevation_service/one_point_per_user.py

```diff
@@ -1,131 +1,164 @@
-
+            break
+    if ranges[-2] < response_time:
+        counters[f"counter{len(ranges)}"] += 1
     total_requests += 1
 
+    # if response_time < percent_ranges[0][1]:
+    #     counters[f"counter{0}"] += 1
+    # elif response_time < percent_ranges[1][1]:
+    #     counters[f"counter{1}"] += 1
+    # else:
+    #     counters[f"counter{2}"] += 1
+    # for index, range_rsp in enumerate(percent_ranges):
+    #     if range_rsp[1] is None and response_time >= range_rsp[0]:
+    #         counters[f"counter{index + 1}"] += 1
+    #         break
+    #     elif range_rsp[0] <= response_time <= range_rsp[1]:
+    #         counters[f"counter{index + 1}"] += 1
+    #         break
+    #
+    # total_requests += 1
+
+
+# @events.request.add_listener
+# def on_request(response_time, **kwargs):
+#     global total_requests, start_time_data
+#     start_time = datetime.datetime.fromtimestamp(kwargs['start_time'])
+#     response_time_data.append(response_time)
+#     start_time_data.append(start_time)
+#     total_requests += 1
+
 
 @events.test_start.add_listener
-def reset_counters(**kwargs):
-    global counters, total_requests, run_number, start_time_data, response_time_data
-    counters = counters
+def reset_data_counters(**kwargs):
+    global total_requests, run_number, counters
     total_requests = 0
     run_number += 1
-    start_time_data = []
-    response_time_data = []
+    counters = initiate_counters_by_ranges(config_ranges=percent_ranges)
 
 
-# Run the Locust test
-class MyUser(CustomUser):
-    wait_time = constant(ElevationConfig.wait_time)
+# @events.test_start.add_listener
+# def reset_counters(**kwargs):
+#     global counters, total_requests
+#     counters = counters
+#     total_requests = 0
+
 
 @events.test_stop.add_listener
 def on_locust_stop(environment, **kwargs):
-    global file_created
-
-    # Ensure file creation code is executed only once
-    if not file_created:
-        percent_value_by_range = {}
-        if total_requests != 0:
-            for index, (key, value) in enumerate(counters.items()):
-                range_percent_val = (value / total_requests) * 100
-                percent_value_by_range[f"{percent_ranges[index]}"] = range_percent_val
-            percent_value_by_range["total_requests"] = total_requests
-            write_rps_percent_results(
-                custom_path=reports_path, percente_value_by_range=percent_value_by_range
-            )
-        else:
-            percent_value_by_range["total_requests"] = total_requests
\ No newline at end of file
+    global total_requests, test_results, counters
+    average_response_time = environment.runner.stats.total.avg_response_time
+    test_results.append(
+        {"users": environment.users_count, "avg_response_time": average_response_time}
+    )
+    create_custom_graph(
+        graph_name="Users_vs_AvgResponseTime",
+        graph_path=reports_path,
+        test_results=test_results,
+        graph_title=None,
```

Comment:
The graph_title parameter of create_custom_graph is set to None. If there is a specifisc title for the graph make it clarity!